### PR TITLE
APIGOV-18136 - set config team name

### PR DESCRIPTION
**Pull request by sbolosan**
@jcollins-axway @kflan-io @vivekschauhan @dfeldick please review.
This is going under the assumption that local config takes precedence over resource config.

### CODE REVIEW

**Review by dfeldick — APPROVED** [review]

**Review by kflan-io — APPROVED** [review]

**Review by kflan-io — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -557,5 +557,14 @@ func applyResConfigToCentralConfig(cfg *config.CentralConfiguration, resCfgAddit
...
 
-	cfg.TeamName = resCfgTeamName
+	// If config team is not blank, use cfg.TeamName
+	// If config team is blank, check resource team name.  If resource team name is not blank, use resource team name
+	// If config team is blank and resource team name is blank, set to "" and let central handle setting team based on organization
+	if cfg.TeamName == "" {
+		if resCfgTeamName != "" {
+			cfg.TeamName = resCfgTeamName
+		} else {
+			cfg.TeamName = ""
```
Comment (jcollins-axway) [inline pkg/agent/agent.go:567]: This is redundant as we know TeamName is already ""

**Review by jcollins-axway — CHANGES_REQUESTED** [review]

File: pkg/agent/agent.go (hunk 1 above)
Comment (sbolosan) [reply pkg/agent/agent.go:567]: true... thanks!

**Review by dfeldick — APPROVED** [review]

**Review by dfeldick — APPROVED** [review]

**Review by kflan-io — APPROVED** [review]

File: pkg/agent/agent.go [hunk 2]
```diff
@@ -557,5 +557,9 @@ func applyResConfigToCentralConfig(cfg *config.CentralConfiguration, resCfgAddit
 		log.GlobalLoggerConfig.Level(logLevel).Apply()
 	}
 
-	cfg.TeamName = resCfgTeamName
+	// If resource team name is not blank, apply it
+	if resCfgTeamName != "" {
```
Comment (jcollins-axway) [inline pkg/agent/agent.go:561]: I still think you want the cfg.TeamName == "" in this condition.

What would happen if both are set, the resource one would take precedence when it should not.

Comment (sbolosan) [reply pkg/agent/agent.go:561]: crap... sorry i got balls in the air again.

**Review by kflan-io — APPROVED** [review]

**Review by dfeldick — APPROVED** [review]

**Review by jcollins-axway — APPROVED** [review]

**Pull request merged**